nexus/db-queries/src/db/datastore/reconfigurator_chicken_switches.rs

```diff
@@ -0,0 +1,351 @@
+            .select(DbReconfiguratorChickenSwitches::as_select())
+            .get_result_async(&*conn)
+            .await
+            .optional()
+            .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?;
+
+        Ok(latest.map(Into::into))
+    }
+
+    /// Insert the current version of the chicken switches in the database
+    ///
+    /// Only succeeds if the prior version is the latest version currently in
+    /// the `reconfigurator_chicken_switches` table. If there are no versions
+    /// currently in the table, then the current swtiches must be at version 1.
+    pub async fn reconfigurator_chicken_switches_insert_latest_version(
+        &self,
+        opctx: &OpContext,
+        switches: &ReconfiguratorChickenSwitches,
+    ) -> Result<(), Error> {
+        if switches.version < 1 {
+            return Err(Error::invalid_request(
+                "version must be greater than 0",
+            ));
+        }
+        opctx
+            .authorize(authz::Action::Modify, &authz::BLUEPRINT_CONFIG)
+            .await?;
+
+        let num_inserted = if switches.version == 1 {
+            self.reconfigurator_chicken_switches_insert_first(opctx, &switches)
+                .await?
+        } else {
+            self.reconfigurator_chicken_switches_insert_next_version(
+                opctx, &switches,
+            )
+            .await?
+        };
+
+        match num_inserted {
+            0 => Err(Error::invalid_request(format!(
+                "version {} is not the most recent",
+                switches.version
+            ))),
+            1 => Ok(()),
+            // This is impossible because we are explicitly inserting only one
+            // row with a unique primary key.
+            _ => unreachable!("query inserted more than one row"),
+        }
+    }
+
+    /// Insert the next version of the chicken switches in the database
+    ///
+    /// Only succeeds if the prior version is the latest version currently
+    /// in the `reconfigurator_chicken_switches` table.
+    ///
+    /// Panics if `switches.version <= 1`;
+    async fn reconfigurator_chicken_switches_insert_next_version(
+        &self,
+        opctx: &OpContext,
+        switches: &ReconfiguratorChickenSwitches,
+    ) -> Result<usize, Error> {
+        assert!(switches.version > 1);
+        let prev_version = switches.version - 1;
+
+        sql_query(
+            r"INSERT INTO reconfigurator_chicken_switches
+                (version, planner_enabled, time_modified)
+              SELECT $1, $2, $3
+              FROM reconfigurator_chicken_switches WHERE version = $4 AND version IN
+              (SELECT version FROM reconfigurator_chicken_switches
+               ORDER BY version DESC LIMIT 1)",
+        )
+        .bind::<sql_types::BigInt, SqlU32>(switches.version.into())
+        .bind::<sql_types::Bool, _>(switches.planner_enabled)
+        .bind::<sql_types::Timestamptz, _>(switches.time_modified)
```

Comment:
Is this letting the caller choose the timestamp on the row? Should we be setting it to `now()` instead? Seems like the kind of thing we should report back but not accept as input.

Reply:
Done in [7531c64](https://github.com/oxidecomputer/omicron/pull/8449/commits/7531c641fdaae07fb8d9af5b657297cd6f117d1c)